src/corefx/System.Globalization.Native/icushim.cpp

```diff
@@ -170,11 +170,31 @@ bool FindLibWithMajorMinorSubVersion(int* majorVer, int* minorVer, int* subVer)
     return false;
 }
 
-// GlobalizationNative_LoadICU
-// This method get called from the managed side during the globalization initialization. 
-// This method shouldn't get called at all if we are running in globalization invariant mode
-// return 0 if failed to load ICU and 1 otherwise
-extern "C" int32_t GlobalizationNative_LoadICU()
+#ifdef OSX_PLATFORM
+
+bool InitializeICULinks(char* symbolName, char* symbolVersion)
+{
+#ifndef OSX_ICU_LIBRARY_PATH
+    static_assert(false, "The ICU Library path is not defined");
+#endif // OSX_ICU_LIBRARY_PATH
+
+    // Usually OSX_ICU_LIBRARY_PATH be "/usr/lib/libicucore.dylib"
+    libicuuc = dlopen(OSX_ICU_LIBRARY_PATH, RTLD_LAZY);
+    
+    if (libicuuc == nullptr)
+    {
+        return false;
+    }
+
+    // in OSX all ICU APIs exist in the same library libicucore.A.dylib 
+    libicui18n = libicuuc;
+
+    return true;
+}
+
+#else
+
+bool InitializeICULinks(char* symbolName, char* symbolVersion)
```

Comment:
I am not good in naming. I used links to include the libraries and symbols. but I can change it as you suggested as I don't have strong opinion on that :-)